prov/efa/src/efa_mr.c

```diff
@@ -372,6 +399,11 @@ static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, void *attr)
 	if (!efa_mr->ibv_mr) {
 		EFA_WARN(FI_LOG_MR, "Unable to register MR: %s\n",
 				fi_strerror(-errno));
+		if (cuda_is_gdrcopy_enabled()) {
+			assert(efa_mr->peer.device.gdrcopy_handle);
```

Comment:
I don't think this assertion is correct. cuda_is_gdrcopy_enabled can return true when attr->iface == FI_HMEM_SYSTEM, which means gdrcopy_handle wouldn't be set.

Reply:
Thank you! added `efa_mr->peer.iface == FI_HMEM_CUDA` to the if condition.